### ISSUE TITLE
ImGuiWindowFlags_AlwaysAutoResize is ignored for docked windows

### ISSUE THREAD

**Issue by aardappel** [ocornut/imgui#6326]
**Version/Branch of Dear ImGui:**

Version: Today (2023-4-12)
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_sdl2.cpp + imgui_impl_opengl3.cpp
Compiler: VS2022
Operating System: Win10

**My Issue/Question:**

I have ImGuiWindowFlags_AlwaysAutoResize set on my windows, and I just enabled docking, however when docked and the content size changes (due to e.g. changing to a different tab) the docked window stays the same width and all content is clipped.

**Comment by ocornut** [comment]
Hello,

This has been discussed several times under various forms. 
IMHO the features are essentially incompatible:

- docking need to apply constraints which is incompatible with the idea of auto-resizing.
- would also affect many other windows potentially the entire screen layout.
- what happens when multiple windows are docked at the same spot? do you expect the docking node and potentially the entire screen layout to start moving around and resizing when clicking on a tab?

I am curious how you would imagine this to work?

**Comment by ocornut** [comment]
I'm afraid those problems are really difficult to solve.

Does it seem much of a problem that your W1/W2 window needs to be resized once by the user?

> Sorry, did search for a combination of ImGuiWindowFlags_AlwaysAutoResize and dock / docking on google and github and came up blank.

I understand it may not be easy to find, at core the problem is: can a window enforce a size in the docking layout. So #6295 #4228 #3363 #2849 #2428 #2386 are essentially a variant of the same thing.

I believe it is desirable to solve this problem for certain uses (e.g. Toolbars, see https://github.com/ocornut/imgui/issues/2648#issuecomment-849741295) but that would probably come with some constraints.
I guess it is unlikely we'll go much further than this toolbar proof-of-concept in the current incarnation of Docking, but it is something I will have in mind for Docking V3.

**Comment by aardappel** [comment]
> I'm afraid those problems are really difficult to solve.

That's fine, I can understand it doesn't fit :)
Floating windows work ok for our use case, docking would have been icing on the cake :)

> Does it seem much of a problem that your W1/W2 window needs to be resized once by the user?

It's not just once. Switch to a different tab and now half of your UI is clipped. Switch back and now your UI contains lots of blank space obscuring your game view unnecessarily. You'd be adjusting all the time, for something ImGui can already do to perfection automatically?

**Comment by ocornut** [comment]
I agree that if there are specific constraints it should be possible, I’ll keep this in mind.

**Comment by learn-more** [comment]
For reference, this is how I handle this 'issue' now in a toy 6502 emulator:

```cpp
void render() override
{
    menu_bar();

    ImGuiID dock = create_dockspace();  // if this returns non-zero, there is no existing dockspace config yet
    ImGuiID left = 0, right = 0, center = 0;
    if (dock)
    {
        ImGuiID tmp;
        ImGui::DockBuilderSplitNode(dock, ImGuiDir_Left, 0.2f, &left, &tmp);
        ImGui::DockBuilderSplitNode(tmp, ImGuiDir_Right, 0.50f, &right, &center);
    }

    if (show_demo)
        ImGui::ShowDemoWindow(&show_demo);

    Registers state = emu.state();

    if (left)
    {
        ImGuiWindowClass cls;
        cls.DockNodeFlagsOverrideSet = ImGuiDockNodeFlags_NoResize;
        ImGui::SetNextWindowClass(&cls);
        ImGui::SetNextWindowDockID(left);
    }
    float width = registers::render(state);
    if (left)
    {
        // Workaround to fake an 'auto-sized' dock node, override the width
        auto dock_node = ImGui::DockBuilderGetNode(left);
        dock_node->Size.x = dock_node->SizeRef.x = width;
    }
    if (right)
        ImGui::SetNextWindowDockID(right);
    memory::render(state.PC, emu.memory());

    if (dock)
        ImGui::DockBuilderFinish(dock);
}
```

The whole dance with the `if (node)` etc is to make sure the user can dock/undock after the initial dock nodes are created.
The fixed size window returns the size it 'wants', and this is then forced on the dock node.
